encfs/StreamNameIO.cpp

```diff
@@ -104,11 +104,13 @@ int StreamNameIO::encodeName(const char *plaintextName, int length,
     encodedName[0] = (mac >> 8) & 0xff;
     encodedName[1] = (mac)&0xff;
     encodeBegin = (unsigned char *)encodedName + 2;
+	rAssert(length <= (bufferLength - 2));
   } else {
     // encfs 0.x stored checksums at the end.
     encodedName[length] = (mac >> 8) & 0xff;
     encodedName[length + 1] = (mac)&0xff;
     encodeBegin = (unsigned char *)encodedName;
+	rAssert(length <= bufferLength);
```

Comment:
You're right, I didn't notice we were writing to length and length+1.  I have fixed this and will push the changes.